From 52b2d0f0496a013548e06a449c8ec8bb4ff00f5f Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 13 Sep 2026 13:36:57 +0200
Subject: [PATCH 03/11] ntdll: Add kombucha_hacks

Adds necessary workarounds/hacks for Roblox Studio and underlying
programs:

1. Fixes versions of Roblox Studio that ship mimalloc from crashing.

2. Fixes issues with WebView2's Vulkan backend not functioning when
   Vinegar's Vulkan layer is loaded.

Signed-off-by: effects3d <effects3dee@gmail.com>
---
 dlls/ntdll/unix/loader.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/dlls/ntdll/unix/loader.c b/dlls/ntdll/unix/loader.c
index b4cedb61c07..8d3a110e55c 100644
--- a/dlls/ntdll/unix/loader.c
+++ b/dlls/ntdll/unix/loader.c
@@ -2291,6 +2291,17 @@ BOOL alert_simulate_sched_quantum;
 BOOL fsync_simulate_sched_quantum;
 BOOL fsync_yield_to_waiters;
 
+static void kombucha_hacks(void)
+{
+    if (!getenv("MIMALLOC_FORCE_REDIRECT")) {
+        setenv("MIMALLOC_FORCE_REDIRECT", "1", 0 );
+    }
+
+    if (strstr(main_argv[1], "msedgewebview2.exe")) {
+        unsetenv( "VK_LOADER_LAYERS_ENABLE" );
+    }
+}
+
 static void hacks_init(void)
 {
     const char *sgi = getenv( "SteamGameId" );
@@ -2453,6 +2464,7 @@ static void start_main_thread(void)
     signal_init_threading();
     dbg_init();
     startup_info_size = server_init_process();
+    kombucha_hacks();
     hacks_init();
     virtual_map_user_shared_data();
     init_cpu_info();
-- 
2.55.0

